src/misc/reachability.c

```diff
@@ -0,0 +1,122 @@
+/* -*- mode: C -*-  */
+/* vim:set ts=4 sw=4 sts=4 et: */
+/*
+   IGraph library.
+   Copyright (C) 2024  The igraph development team <igraph@igraph.org>
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "igraph_adjlist.h"
+#include "igraph_community.h"
+#include "igraph_components.h"
+#include "igraph_interface.h"
+#include "igraph_reachability.h"
+
+void _print_vector_int(const igraph_vector_int_t *v) {
```

Comment:
I guess this function was meant for debugging purposes. We have `igraph_vector_int_print()` so you can use that. This function has to be removed because it refers to `printf()` and the R interface maintainers will complain. R has very strict rules about what symbols may appear in extension modules and `printf()` is not allowed.

Reply:
Removed